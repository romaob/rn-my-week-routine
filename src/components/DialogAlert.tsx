import React from 'react';
import {View, Image, StyleSheet, Modal} from 'react-native';
import Label, {FontSize} from './Label';
import Button, {ButtonColorType, ButtonSize} from './Button';
import {colors} from '../values/colors';
import {sizes} from '../values/sizes';
import useString from '../hooks/useString';

interface DialogAlertProps {
  show: boolean;
  title: string;
  message: string;
  onCancel: () => void;
  onConfirm: () => void;
}

const DialogAlert = ({
  show,
  title,
  message,
  onCancel,
  onConfirm,
}: DialogAlertProps): JSX.Element | null => {
  const {getString} = useString();
  if (!show) {
    return null;
  } else {
    return (
      <View style={styles.backdrop}>
        <Modal
          visible={show}
          animationType="fade"
          transparent
          testID="dialogAlert">
          <View style={styles.centeredView}>
            <View style={styles.container}>
              <View style={styles.topRow}>
                <View style={styles.cancelIcon}>
                  <Image
                    style={styles.icon}
                    source={require('../assets/images/cancel.png')}
                    testID="dialogAlertIcon"
                  />
                </View>
                <Label
                  text={title}
                  size={FontSize.LARGE}
                  color={colors.light.danger}
                  testID="dialogAlertTitle"
                />
              </View>
              <View style={styles.body}>
                <Label text={message} testID="dialogAlertMessage" />
              </View>
              <View style={styles.bottomRow}>
                <Button
                  label={getString('cancel')}
                  size={ButtonSize.LARGE}
                  onPress={onCancel}
                  testID="dialogAlertCancelButton"
                />
                <Button
                  label={getString('delete').toUpperCase()}
                  size={ButtonSize.LARGE}
                  onPress={onConfirm}
                  colorType={ButtonColorType.DANGER}
                  testID="dialogAlertConfirmButton"
                />
              </View>
            </View>
          </View>
        </Modal>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  backdrop: {
    display: 'flex',
    position: 'absolute',
    width: '120%',
    height: '120%',
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
  },
  container: {
    backgroundColor: colors.light.cardBackground,
    borderRadius: 10,
    padding: sizes.padding.lg,
  },
  topRow: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  cancelIcon: {
    backgroundColor: 'red',
    width: 30,
    height: 30,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 10,
  },
  body: {
    marginTop: 20,
    marginBottom: 20,
  },
  bottomRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  icon: {
    width: 20,
    height: 20,
  },
});

export default DialogAlert;
