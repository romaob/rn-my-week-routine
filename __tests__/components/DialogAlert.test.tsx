import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import DialogAlert from '../../src/components/DialogAlert';

const TEST_TITLE = 'Test Title';
const TEST_MESSAGE = 'Test Message';
const onConfirmTest = jest.fn();
const onCancelTest = jest.fn();

describe('DialogAlert render tests', () => {
  it('should render correctly', () => {
    const result = render(
      <DialogAlert
        show
        title={TEST_TITLE}
        message={TEST_MESSAGE}
        onCancel={() => {}}
        onConfirm={() => {}}
      />,
    );
    //Main test
    expect(result).toBeTruthy();
    //If it matches the snapshot
    expect(result).toMatchSnapshot();
    //If the components are rendered
    expect(result.getByTestId('dialogAlert')).toBeTruthy();
    expect(result.getByTestId('dialogAlertTitle')).toBeTruthy();
    expect(result.getByTestId('dialogAlertMessage')).toBeTruthy();
    expect(result.getByTestId('dialogAlertCancelButton')).toBeTruthy();
    expect(result.getByTestId('dialogAlertConfirmButton')).toBeTruthy();
    expect(result.getByTestId('dialogAlertIcon')).toBeTruthy();
  });

  it('should render the provided title', () => {
    const result = render(
      <DialogAlert
        show
        title={TEST_TITLE}
        message={TEST_MESSAGE}
        onCancel={() => {}}
        onConfirm={() => {}}
      />,
    );
    expect(result).toBeTruthy();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('dialogAlert')).toBeTruthy();
    expect(result.getByTestId('dialogAlertTitle')).toBeTruthy();
    //Check if the title contain the text inside
    expect(
      result
        .getByTestId('dialogAlertTitle')
        .props.children.some((child: any) => child === TEST_TITLE),
    ).toBeTruthy();
  });

  it('should render the provided message', () => {
    const result = render(
      <DialogAlert
        show
        title={TEST_TITLE}
        message={TEST_MESSAGE}
        onCancel={() => {}}
        onConfirm={() => {}}
      />,
    );
    expect(result).toBeTruthy();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('dialogAlert')).toBeTruthy();
    expect(result.getByTestId('dialogAlertMessage')).toBeTruthy();
    //Check if the body text contain the text inside
    expect(
      result
        .getByTestId('dialogAlertMessage')
        .props.children.some((child: any) => child === TEST_MESSAGE),
    ).toBeTruthy();
  });
});

describe('DialogAlert behavior tests', () => {
  it('should be hidden based on the show prop', () => {
    const result = render(
      <DialogAlert
        show={false}
        title={TEST_TITLE}
        message={TEST_MESSAGE}
        onCancel={() => {}}
        onConfirm={() => {}}
      />,
    );
    expect(result.queryByTestId('dialogAlert')).toBeFalsy();
  });

  it('should call the onCancel callback when the cancel button is pressed', () => {
    const result = render(
      <DialogAlert
        show
        title={TEST_TITLE}
        message={TEST_MESSAGE}
        onCancel={onCancelTest}
        onConfirm={onConfirmTest}
      />,
    );
    expect(result).toBeTruthy();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('dialogAlert')).toBeTruthy();
    expect(result.getByTestId('dialogAlertCancelButton')).toBeTruthy();

    //Press the cancel button
    fireEvent.press(result.getByTestId('dialogAlertCancelButton'));
    expect(onCancelTest).toHaveBeenCalled();
  });

  it('should call the onConfirm callback when the confirm button is pressed', () => {
    const result = render(
      <DialogAlert
        show
        title={TEST_TITLE}
        message={TEST_MESSAGE}
        onCancel={onCancelTest}
        onConfirm={onConfirmTest}
      />,
    );
    expect(result).toBeTruthy();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('dialogAlert')).toBeTruthy();
    expect(result.getByTestId('dialogAlertConfirmButton')).toBeTruthy();

    //Press the confirm button
    fireEvent.press(result.getByTestId('dialogAlertConfirmButton'));
    expect(onConfirmTest).toHaveBeenCalled();
  });
});
