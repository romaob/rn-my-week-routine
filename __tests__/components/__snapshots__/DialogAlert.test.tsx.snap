// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogAlert behavior tests should call the onCancel callback when the cancel button is pressed 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="dialogAlert"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 20,
                "height": 30,
                "justifyContent": "center",
                "marginRight": 10,
                "width": 30,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/images/cancel.png",
                }
              }
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              testID="dialogAlertIcon"
            />
          </View>
          <Text
            style={
              {
                "color": "#f44336",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="dialogAlertTitle"
          >
            Test Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 14,
              }
            }
            testID="dialogAlertMessage"
          >
            Test Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#3a59db",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertCancelButton"
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f44336",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertConfirmButton"
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogAlert behavior tests should call the onConfirm callback when the confirm button is pressed 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="dialogAlert"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 20,
                "height": 30,
                "justifyContent": "center",
                "marginRight": 10,
                "width": 30,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/images/cancel.png",
                }
              }
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              testID="dialogAlertIcon"
            />
          </View>
          <Text
            style={
              {
                "color": "#f44336",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="dialogAlertTitle"
          >
            Test Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 14,
              }
            }
            testID="dialogAlertMessage"
          >
            Test Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#3a59db",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertCancelButton"
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f44336",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertConfirmButton"
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogAlert render tests should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="dialogAlert"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 20,
                "height": 30,
                "justifyContent": "center",
                "marginRight": 10,
                "width": 30,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/images/cancel.png",
                }
              }
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              testID="dialogAlertIcon"
            />
          </View>
          <Text
            style={
              {
                "color": "#f44336",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="dialogAlertTitle"
          >
            Test Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 14,
              }
            }
            testID="dialogAlertMessage"
          >
            Test Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#3a59db",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertCancelButton"
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f44336",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertConfirmButton"
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogAlert render tests should render the provided message 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="dialogAlert"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 20,
                "height": 30,
                "justifyContent": "center",
                "marginRight": 10,
                "width": 30,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/images/cancel.png",
                }
              }
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              testID="dialogAlertIcon"
            />
          </View>
          <Text
            style={
              {
                "color": "#f44336",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="dialogAlertTitle"
          >
            Test Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 14,
              }
            }
            testID="dialogAlertMessage"
          >
            Test Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#3a59db",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertCancelButton"
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f44336",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertConfirmButton"
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`DialogAlert render tests should render the provided title 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "display": "flex",
      "height": "120%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "120%",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="dialogAlert"
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "padding": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 20,
                "height": 30,
                "justifyContent": "center",
                "marginRight": 10,
                "width": 30,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/images/cancel.png",
                }
              }
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
              testID="dialogAlertIcon"
            />
          </View>
          <Text
            style={
              {
                "color": "#f44336",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 24,
              }
            }
            testID="dialogAlertTitle"
          >
            Test Title
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "display": "flex",
                "flexDirection": "row",
                "fontSize": 14,
              }
            }
            testID="dialogAlertMessage"
          >
            Test Message
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#3a59db",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertCancelButton"
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f44336",
                "borderRadius": 5,
                "display": "flex",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="dialogAlertConfirmButton"
          />
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
